common_code/inspiral/CE_orbit.f90

```diff
@@ -76,66 +91,163 @@ subroutine CE_orbit_adjust(id, ierr)
-         ! Calculate the energies
-         E_init = -standard_cgrav * CE_companion_mass * Msun * M_encl / (2.0 * CE_companion_position * Rsun)
-         E_loss = CE_energy_rate * s% dt
-         E_final = E_init - E_loss
-
-         ! Move from outside of star in to find cell containing companion
-         E_tmp = 0d0
-         k = 1
-         do while (E_tmp > E_final)
-            M_inner = s% m(k)
-            R_inner = s% r(k)
-            E_tmp = -standard_cgrav * CE_companion_mass * Msun * M_inner / (2.0 * R_inner)
-            k = k + 1
-         end do
-
-         ! If companion is outside star, set k to 3
-         if (k < 3) k=3
-
-         ! save end points of cell containing companion
-         M_inner = s% m(k-2)
-         R_inner = s% r(k-2)
-         M_outer = s% m(k-1)
-         R_outer = s% r(k-1)
-
-         ! We could choose to interpolate for R using M as the independent variable. Instead, here we
-         ! linearly interpolate across cell (using k as the independent variable)
-         M_slope = (M_outer - M_inner) / real((k-1) - (k-2))
-         M_int = s% m(k-1) - M_slope * real(k-1)
-         R_slope = (R_outer - R_inner) / real((k-1) - (k-2))
-         R_int = s% r(k-1) - R_slope * real(k-1)
-
-         ! Given the final energy, E_final, determine the resulting k that solves the equation
-         top = 2.0 * E_final * R_int + standard_cgrav * CE_companion_mass * Msun * M_int
-         bottom = 2.0 * E_final * R_slope + standard_cgrav * CE_companion_mass * Msun * M_slope
-         k_final = -top / bottom
-
-         ! Now use the interpolations and the derived k_final, determine the resulting separation and enclosed mass
-         R_final = R_slope * k_final + R_int
-         M_final = M_slope * k_final + M_int
-
-         s% xtra(2) = R_final/Rsun
-         !Saving as s% xtra(9) the enclosed mass so that we output it in the history data
-         s% xtra(9) = M_final/Msun
-
-         ! Calculate the angular momentum lost to the star's envelope
-         J_tmp = (CE_companion_mass * Msun)**2 * M_final**2 / (CE_companion_mass * Msun + M_final)
-         J_final = sqrt(standard_cgrav * J_tmp * R_final)
-
-         !The angular momentum that is lost from the orbit of the companion
-         ! is added to the envelope of the donor.
-         orbital_ang_mom_lost = J_final - J_init
-         !We save in s% xtra(6) the total torque that will be applied to the Envelope
-         s% xtra(6) = max(0.0d0, -orbital_ang_mom_lost/s% dt) 
-
-         ! Keep track of orbital energy and angular momentum
-         s% xtra(8) = E_final
-         s% xtra(10) = J_final
+         if (s% x_logical_ctrl(7)) then
+            ! Complete integration of the eqs. of motion
+
+            if (s% model_number .eq. 1) then
+               ! Determine the initial conditions of the velocity
+               if (s% x_logical_ctrl(8)) then
+                  s% xtra(25) = s% x_ctrl(2) * s% r(1) / Rsun
```

Comment:
Why do you write the radial position into the variable for the angular position `s% xtra(25)`?